module_2/gifts/service/src/main/java/com/epam/esm/service/validation/CertificateValidation.java

```diff
@@ -0,0 +1,248 @@
+		return errors;
+	}
+
+	private Map<ErrorCode, String> validateAllCertificateUpdatableNotNullFields(CertificateDto certificateDto) {
+
+		Map<ErrorCode, String> errors = new HashMap<>();
+
+		String name = certificateDto.getName();
+		if (name != null && !validateName(name)) {
+			errors.put(ErrorCode.INVALID_CERTIFICATE_NAME,
+					EntityConstant.NAME + Util.DELIMITER + certificateDto.getName());
+		}
+
+		String description = certificateDto.getDescription();
+		if (description != null && !validateDescription(description)) {
+			errors.put(ErrorCode.INVALID_CERTIFICATE_DESCRIPTION,
+					EntityConstant.CERTIFICATE_DESCRIPTION + Util.DELIMITER + certificateDto.getDescription());
+		}
+
+		BigDecimal price = certificateDto.getPrice();
+		if (price != null && !validatePrice(price)) {
+			errors.put(ErrorCode.INVALID_CERTIFICATE_PRICE,
+					EntityConstant.CERTIFICATE_PRICE + Util.DELIMITER + certificateDto.getPrice());
+		}
+
+		Integer duration = certificateDto.getDuration();
+		if (duration != null && !validateDuration(duration)) {
+			errors.put(ErrorCode.INVALID_CERTIFICATE_DURATION,
+					EntityConstant.CERTIFICATE_DURATION + Util.DELIMITER + certificateDto.getDuration());
+		}
+		return errors;
+	}
+
+	public void validateReadParams(MultiValueMap<String, String> params) {
+		Util.checkNull(params);
+		if (!POSSIBLE_READ_PARAMS.containsAll(params.keySet())) {
+			throw new ValidationException(EntityConstant.PARAMS + Util.DELIMITER + params,
+					ErrorCode.INVALID_CERTIFICATE_READ_PARAM);
+		}
+
+		if (params.containsKey(EntityConstant.ORDER)) {
+			if (!POSSIBLE_SORT_FIELD.containsAll(params.get(EntityConstant.ORDER))) {
+				throw new ValidationException(EntityConstant.PARAMS + Util.DELIMITER + params,
+						ErrorCode.INVALID_CERTIFICATE_SORT_PARAM);
+			}
+		}
+		if (params.containsKey(EntityConstant.PAGE)) {
+			int page = DEFAULT_PAGE_NUMBER;
+			String initialPage = params.get(EntityConstant.PAGE).get(0);
+			try {
+				page = Integer.parseInt(initialPage);
+			} catch (NumberFormatException e) {
+				throw new ValidationException(EntityConstant.PAGE + Util.DELIMITER + initialPage,
+						ErrorCode.INVALID_PAGE_FORMAT);
+			}
+			if (page <= 0) {
+				throw new ValidationException(EntityConstant.PAGE + Util.DELIMITER + page,
+						ErrorCode.NEGATIVE_PAGE_NUMBER);
+			}
+		} else {
+			params.put(EntityConstant.PAGE, Arrays.asList(Integer.toString(DEFAULT_PAGE_NUMBER)));
+		}
+
+		if (params.containsKey(EntityConstant.LIMIT)) {
+			int limit = OFFSET;
+			try {
+				limit = Integer.parseInt(params.get(EntityConstant.LIMIT).get(0));
+			} catch (NumberFormatException e) {
+				throw new ValidationException(EntityConstant.LIMIT + Util.DELIMITER + limit,
+						ErrorCode.INVALID_OFFSET_FORMAT);
+			}
+			if (limit <= 0) {
+				throw new ValidationException(EntityConstant.LIMIT + Util.DELIMITER + limit, ErrorCode.NEGATIVE_OFFSET);
+			}
+		} else {
+			params.put(EntityConstant.LIMIT, Arrays.asList(Integer.toString(OFFSET)));
+		}
+	}
```

Comment:
Throw exception it's a not responsibility of validation class. Methods of validation class must only check if the parameter meets the conditions or no, but throw an exception it's an obligation of caller. Do not touch it for now, you need a lot of time to refactor this, but keep in mind